### ISSUE TITLE
ELM namelist confusion:  hgrid vs res

### ISSUE THREAD

**Issue by mt5555** [E3SM-Project/E3SM#3464]
Per "namelist_defaults_clm4_5.xml",
"The recognized attributes are: hgrid, defaults, mask, ic_ymd, ic_tod, sim_year and all configuration attributes from the config_cache.xml file (with keys converted to upper-case)."

And thus there are a lot of defaults with the hgrid keyword, for example:
```
<fsurdat hgrid="ne30np4"     sim_year="2000" >                                                                                                                 
lnd/clm2/surfdata_map/surfdata_ne30np4_simyr2000_c190730.nc</fsurdat>
```
Note that internally, in CLM's "configure" script, it is using the variable "res".  The variables that can be used to parse the XML files are explicitly set by configure, and the XML variable "hgrid"
is set many times based on configure's "res".

The one except to this is for use cases. For use cases, the XMl variables are initialized in CLMBuildNamelist.pm , in subroutine process_namelist_commandline_use_case
with code like this:

```
 my %settings;
    $settings{'res'}            = $nl_flags->{'res'};
    $settings{'rcp'}            = $nl_flags->{'rcp'};
    $settings{'mask'}           = $nl_flags->{'mask'};
    $settings{'sim_year'}       = $nl_flags->{'sim_year'};
    $settings{'sim_year_range'} = $nl_flags->{'sim_year_range'};
    $settings{'phys'}           = $nl_flags->{'phys'};
....
```

This means that when parsing XML use cases, the grid keyword "hgrid" will not be set and "res" must be used instead.  This for example, in 20thC_CMIP6_transient.xml we have:
```
<fsurdat res="r05">lnd/clm2/surfdata_map/surfdata_0.5x0.5_simyr1850_c190418.nc</fsurdat>
```

In summary, the perl code uses the variable "res", while the XML code not related to use cases uses the variable "hgrid", and the XML code for use_cases uses the variable "res".   For all other variables (like mask and sim_year), the name used by the XML and the name used in the perl code are the same.

It might be nice to change all uses of "hgrid" to "res", or vice versa.   But this would touch a lot of code.  Failing that, I think at the least the use_cases XML should be changed to use "hgrid" so it is consistent with the usage in the namelist_defaults XML.  

I will add this to PR #3459   if there are no objections

**Comment by tangq** [comment]
@mt5555 , will your proposed changes of using "hgrid" in both XML and use_cases solve some of the issues reported by https://github.com/E3SM-Project/E3SM/issues/3453?

Currently, some of the land default settings (e.g., finidat and fsurdat) have to be set in namelist_defaults_clm4_5.xml. With this change, the cases using different use_case xml files can set different land values. However, when the land "hgrid" and use_case file are the same (for example, different North America and Antarctica RRMs running with F2010SC5-CMIP6 compset), it still cannot use different land initial conditions by default.

**Comment by mt5555** [comment]
I dont think #3453 can help.  If just fixes an inconsistency (confusion over when was should use "hgrid=" or "res=").

**Comment by mt5555** [comment]
proposed change was implemented in PR #3459
